src/main/java/com/example/gildedrose/controller/AgedBrieItemController.java

```diff
@@ -0,0 +1,22 @@
+package com.example.gildedrose.controller;
+
+import com.example.gildedrose.item.AgedBrieItem;
+import com.example.gildedrose.item.Updateable;
+import com.example.gildedrose.service.ItemRepository;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class AgedBrieItemController {
```

Comment:
**question**: It seems like we would need to add a new endpoint for every special item that we add to the system. Is there way we could make this more open/closed to the types of items that exist in the system?

Reply:
Somehow I thought I was following open closed principle by adding a new controller (i.e. class) for any addition of a new product.  But as you suggest, there must be a better way.  I thought of a wildcard at the end of the url example items/{name}?  But then we would need some way to map the wildcard to the item class.

Reply:
What if clients posted to `/items` and the name was passed in a part of the request body? 

Something to keep in mind is that the request body object doesn't have to be the same as the domain model. It is perfectly acceptable to create a new object to represent the JSON request/response that is independent of you `Updatable` item classes.

Reply:
That is also an option.  I thought that would be error prone as the name could be entered incorrectly

Reply:
It certainly could be entered incorrectly, and that is probably an outcome you would want to plan for. However, that feels like it would be the responsibility for the validation layer.

The validation layer right now is built into the way you built the endpoints, and it works if you only want to validate the name. But it is not going to work so well for validating the sell in or the initial quality.

So, if you want something that is going to be flexible to those potential futures, a dedicated validation layer is probably going to serve you better than baking it into the routes.